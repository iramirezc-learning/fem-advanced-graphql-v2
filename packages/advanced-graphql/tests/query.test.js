const createTestServer = require('./helper')
const { FEED, ME } = require('./queries')

const mockUser = {
  id: 1,
  email: 'user@example.com',
  avatar: 'http://placeholder.com',
  verified: true,
  createdAt: new Date(2021, 2, 11).getTime(),
  role: 'MEMBER'
}

describe('queries', () => {
  test('feed', async () => {
    const { query } = createTestServer({
      user: mockUser,
      models: {
        Post: {
          findMany: jest.fn(() => [
            {
              id: 1,
              message: 'hello',
              createdAt: 12345839,
              likes: 20,
              views: 300
            }
          ])
        }
      }
    })

    const res = await query({ query: FEED })
    expect(res).toMatchSnapshot()
  })

  test('me', async () => {
    const { query } = createTestServer({
      user: mockUser,
      models: {
        Settings: {
          findOne: jest.fn(() => ({
            theme: 'DARK',
            emailNotifications: true,
            pushNotifications: false
          }))
        }
      }
    })

    const res = await query({ query: ME })
    expect(res).toMatchSnapshot()
  })
})
