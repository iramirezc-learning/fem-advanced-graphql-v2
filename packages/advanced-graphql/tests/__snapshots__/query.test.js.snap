// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries feed 1`] = `
Object {
  "data": Object {
    "feed": Array [
      Object {
        "createdAt": "12345839",
        "id": "1",
        "likes": 20,
        "message": "hello",
        "views": 300,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`queries me 1`] = `
Object {
  "data": Object {
    "me": Object {
      "avatar": "http://placeholder.com",
      "createdAt": "1615442400000",
      "email": "user@example.com",
      "id": "1",
      "role": "MEMBER",
      "settings": Object {
        "emailNotifications": true,
        "pushNotifications": false,
        "theme": "DARK",
      },
      "verified": true,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
